Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift

```diff
@@ -0,0 +1,143 @@
+                Multiline asterisk comment
+            */
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            //↓Something
+            """),
+            Example("""
+            //↓MARK
+            """),
+            Example("""
+            //↓👨‍👨‍👦‍👦Something
+            """),
+            Example("""
+            func a() {
+                //↓This needs refactoring
+                print("Something")
+            }
+            //↓We should improve above function
+            """),
+            Example("""
+            ///↓This is a comment
+            """),
+            Example("""
+            /// Multiline triple-slash
+            ///↓This line is incorrect, though
+            """),
+            Example("""
+            //↓- MARK: Mark comment
+            """)
+        ],
+        corrections: [
+            Example("//↓Something"): Example("// Something"),
+            Example("//↓- MARK: Mark comment"): Example("// - MARK: Mark comment"),
+            Example("""
+            /// Multiline triple-slash
+            ///↓This line is incorrect, though
+            """): Example("""
+            /// Multiline triple-slash
+            /// This line is incorrect, though
+            """),
+            Example("""
+            func a() {
+                //↓This needs refactoring
+                print("Something")
+            }
+            //↓We should improve above function
+            """): Example("""
+            func a() {
+                // This needs refactoring
+                print("Something")
+            }
+            // We should improve above function
+            """)
+        ]
+    )
+
+    public func violationRanges(in file: SwiftLintFile) -> [NSRange] {
+        // Find all comment tokens in the file and regex search them for violations
+        let commentTokens = file.syntaxMap.tokens.filter { [.comment, .docComment].contains($0.kind) }
+        return commentTokens.compactMap { (token: SwiftLintSyntaxToken) -> [NSRange]? in
+            guard let commentBody = file.stringView.substringWithByteRange(token.range).map(StringView.init) else {
+                return nil
+            }
+            // Look for 2-3 slash characters followed immediately by a non-whitespace, non-slash
+            // character (this is a violation)
+            return regex("^(\\/){2,3}[^\\s\\/]").matches(in: commentBody, options: .anchored)
+                .compactMap { result in
+                    // Set the location to be directly before the first non-slash,
+                    // non-whitespace character which was matched
+                    guard let characterRange = file.stringView.byteRangeToNSRange(
+                        ByteRange(
+                            location: ByteCount(
+                                token.range.lowerBound.value + result.range.upperBound - 1
+                            ),
+                            length: ByteCount(1)
```

Comment:
Why is this 1 rather than 0? Also ByteCount is integer literal expressible:

```suggestion
                            length: 0
```

Reply:
I originally thought it should be `1` since the autocorrection looks at the characters and inserts those characters prepending a space, but it occurs to me that we could probably insert a space just fine with 0 length. Will update!